### DISCUSSION TITLE
Is it necessary to provide a new method to append the query string?like kong.service.request.add_query(newquerystring)

### DISCUSSION THREAD

**Discussion by baofhuang** [Kong/kong#10326] [Ideas and feature requests]
kong.service.request.set_query(args) will discard the original query string and only append the new query string.sometimes we only need to append the new query string,keeping the the original query string

**Comment by ghost** [comment]
> we only need to append the new query string

Even if this is the case, there is no direct add_query API in OpenResty, the implementation in Kong would look like the following steps:

1. kong.service.request.get_query(args)
2. append new args
3. kong.service.request.set_query(new args)
